src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -1212,6 +1211,7 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
 
         // Remove Pad in front of MaxPool if both the pads_begin and pads_end are zero.
         manager.register_pass<ov::pass::EliminatePad>();
+        manager.register_pass<ov::pass::ConvertNMS9ToNMSIEInternal>();
```

Comment:
As discussed offline, I think it would just hide the issue. Better approach is to fix the dynamism of NMS operator.